module/zfs/dmu.c

```diff
@@ -823,17 +823,12 @@ dmu_free_range(objset_t *os, uint64_t object, uint64_t offset,
 }
 
 int
-dmu_read(objset_t *os, uint64_t object, uint64_t offset, uint64_t size,
-    void *buf, uint32_t flags)
+dmu_read_impl(dnode_t *dn, uint64_t offset, uint64_t size,
+    void *buf, uint32_t flags, void *tag)
```

Comment:
I don't think you need the `tag` argument.  This func can continue holding/releasing the dbufs using FTAG, because they are held/released by this function and not any others.